hyprtester/src/tests/main/window.cpp

```diff
@@ -131,6 +131,123 @@ static void testSwapWindow() {
     EXPECT(Tests::windowCount(), 0);
 }
 
+static bool expectActiveWindow(const std::string& class_, char fullscreen) {
+    std::string activeWin     = getFromSocket("/activewindow");
+    auto        winClass      = getWindowAttribute(activeWin, "class: ");
+    auto        winFullscreen = getWindowAttribute(activeWin, "fullscreen: ");
+    if (winClass == class_ && winFullscreen[0] == fullscreen)
+        return true;
+    else {
+        NLog::log("{}Wrong active window: expected class {} fullscreen '{}', found class {}, fullscreen '{}'", Colors::RED, class_, fullscreen, winClass, winFullscreen);
+        return false;
+    }
+}
+
+/// Tests behavior of a window being focused when on that window's workspace
+/// another fullscreen window exists.
+static bool testWindowFocusOnFullscreenConflict() {
```

Comment:
I just noticed that there are already some tests for `misc:new_window_takes_over_fullscreen` (which is now `misc:on_focus_under_fullscreen`) in tests/main/misc.cpp. Should I do something about it? Should we keep all the tests for this at the same place and in the same style?

Reply:
probably just change the variable name to make the tests pass. Add new tests for the new functionality.